flink-runtime/src/test/java/org/apache/flink/runtime/operators/AbstractOuterJoinTaskTest.java

```diff
@@ -0,0 +1,501 @@
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+
+		try {
+			addInput(new UniformIntTupleGenerator(keyCnt1, valCnt1, true), serializer);
+			addInputSorted(new UniformIntTupleGenerator(keyCnt2, valCnt2, false), serializer, this.comparator2.duplicate());
+			testDriver(testTask, MockJoinStub.class);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
+		}
+
+		final int expCnt = calculateExpectedCount(keyCnt1, valCnt1, keyCnt2, valCnt2);
+
+		Assert.assertTrue("Resultset size was "+this.outList.size()+". Expected was "+expCnt, this.outList.size() == expCnt);
+
+		this.outList.clear();
+	}
+
+	@Test
+	public void testMergeOuterJoinTask() {
+		int keyCnt1 = 20;
+		int valCnt1 = 20;
+
+		int keyCnt2 = 20;
+		int valCnt2 = 20;
+
+		setOutput(this.outList, this.serializer);
+		addDriverComparator(this.comparator1);
+		addDriverComparator(this.comparator2);
+		getTaskConfig().setDriverPairComparator(new RuntimePairComparatorFactory());
+		getTaskConfig().setDriverStrategy(driverStrategy);
+		getTaskConfig().setRelativeMemoryDriver(bnljn_frac);
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+
+		addInput(new UniformIntTupleGenerator(keyCnt1, valCnt1, true), serializer);
+		addInput(new UniformIntTupleGenerator(keyCnt2, valCnt2, true), serializer);
+
+		try {
+			testDriver(testTask, MockJoinStub.class);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
+		}
+
+		final int expCnt = calculateExpectedCount(keyCnt1, valCnt1, keyCnt2, valCnt2);
+
+		Assert.assertTrue("Resultset size was "+this.outList.size()+". Expected was "+expCnt, this.outList.size() == expCnt);
+
+		this.outList.clear();
+	}
+
+	@Test
+	public void testFailingOuterJoinTask() {
+		int keyCnt1 = 20;
+		int valCnt1 = 20;
+
+		int keyCnt2 = 20;
+		int valCnt2 = 20;
+
+		setOutput(new DiscardingOutputCollector<Tuple2<Integer, Integer>>());
+		addDriverComparator(this.comparator1);
+		addDriverComparator(this.comparator2);
+		getTaskConfig().setDriverPairComparator(new RuntimePairComparatorFactory());
+		getTaskConfig().setDriverStrategy(driverStrategy);
+		getTaskConfig().setRelativeMemoryDriver(bnljn_frac);
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+
+		addInput(new UniformIntTupleGenerator(keyCnt1, valCnt1, true), serializer);
+		addInput(new UniformIntTupleGenerator(keyCnt2, valCnt2, true), serializer);
+
+		try {
+			testDriver(testTask, MockFailingJoinStub.class);
+			Assert.fail("Driver did not forward Exception.");
+		} catch (ExpectedTestException e) {
+			// good!
```

Comment:
For that you can use the `@Test(expected = ExpectedTestException.class)` annotation.